plugins/package-managers/maven/src/test/kotlin/utils/DependencyTreeParserTest.kt

```diff
@@ -0,0 +1,376 @@
+            val remoteRepositories = listOf(remoteRepo1, remoteRepo2)
+            val project = spyk(createProject("module1", "eclipse-plugin")) {
+                every { remoteProjectRepositories } returns remoteRepositories
+            }
+
+            val projectNode = DependencyTreeMojoNode(
+                "org.ossreviewtoolkit",
+                "module1",
+                "1.2.3-SNAPSHOT",
+                "eclipse-plugin",
+                "",
+                "",
+                listOf(
+                    DependencyTreeMojoNode(
+                        "org.apache.commons",
+                        "commons-configuration2",
+                        "2.11.0",
+                        "jar",
+                        "compile",
+                        ""
+                    ),
+                    DependencyTreeMojoNode(
+                        "org.apache.commons",
+                        "commons-lang3",
+                        "3.14.0",
+                        "jar",
+                        "compile",
+                        ""
+                    )
+                )
+            )
+
+            val dependencies = parseDependencyTree(
+                inputStreamFor(projectNode),
+                listOf(project)
+            ).single()
+
+            dependencies.children.forAll { node ->
+                node.repositories shouldContainExactly remoteRepositories
+            }
+        }
+
+        "skip unknown projects" {
+            val projectNode1 = DependencyTreeMojoNode(
+                "org.ossreviewtoolkit",
+                "module1",
+                "1.2.3-SNAPSHOT",
+                "pom",
+                "",
+                "",
+                listOf(
+                    DependencyTreeMojoNode(
+                        "org.apache.commons",
+                        "commons-configuration2",
+                        "2.11.0",
+                        "jar",
+                        "compile",
+                        ""
+                    )
+                )
+            )
+            val projectNode2 = DependencyTreeMojoNode(
+                "org.ossreviewtoolkit",
+                "module2",
+                "1.2.3-SNAPSHOT",
+                "eclipse-plugin",
+                "",
+                "",
+                listOf(
+                    DependencyTreeMojoNode(
+                        "org.apache.commons",
+                        "commons-lang3",
+                        "3.14.0",
+                        "jar",
+                        "compile",
+                        ""
+                    )
+                )
+            )
+
+            val projectDependencies = parseDependencyTree(
+                inputStreamFor(projectNode1, projectNode2),
+                listOf(createProject("module1"))
+            ).toList()
+
+            projectDependencies shouldHaveSize 1
+            compareNodes(projectNode1, projectDependencies.first())
```

Comment:
Alternatively:

            projectDependencies.shouldBeSingleton {
                compareNodes(projectNode, it)
            }